h5py/_hl/group.py

```diff
@@ -134,10 +134,20 @@ def create_dataset(self, name, shape=None, dtype=None, data=None, **kwds):
             kwds['track_order'] = h5.get_config().track_order
 
         with phil:
-            dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)
+            group = self
+            if name:
+                if '/' in name:
+                    h5objects = [obj for obj in name.split('/') if len(obj)]
+                    name = h5objects[-1]
+                    h5objects = h5objects[:-1]
+
+                    for new_group in h5objects:
+                        group = self.get(new_group) or group.create_group(new_group)
```

Comment:
Should this be `group.get(new_group)` (instead of `self...`)?

Reply:
Also, I think you can use `self.require_group()` to create a group if it doesn't exist. That also does the iteration for nested levels, so you don't need the loop.

Reply:
> Should this be `group.get(new_group)` (instead of `self...`)?

Totally right, just fixed.


> Also, I think you can use `self.require_group()` to create a group if it doesn't exist. That also does the iteration for nested levels, so you don't need the loop.

Does it work also for absolute paths or is it just creating subgroups under the group the method is invoked on? The tests are failing if I use  `require_group`

Reply:
> Does it work also for absolute paths or is it just creating subgroups under the group the method is invoked on?

Looking at the code I think it should work, but maybe something isn't handling absolute paths properly. If it's not easy to fix, we can leave it with the iteration for this PR and open an issue to investigate it.

Reply:
I can check it when I will have time. The easiest way to check is to write some tests for ` require_group`.  I would propose to close this PR and I will work on it separately in an issue assigned to me.

Reply:
I think you changed to `group.get()` below, but there's still a `self.get()` on this line.